editor/editor_resource_picker.cpp

```diff
@@ -159,6 +159,15 @@ void EditorResourcePicker::_file_selected(const String &p_path) {
 		}
 
 		if (!any_type_matches) {
+			if (!loaded_resource->get_path().ends_with(".txt")) {
+				for (Ref<TextFile> file : ScriptEditor::get_singleton()->get_open_text_files()) {
+					if (file.ptr()->get_path() != loaded_resource->get_path()) {
+						continue;
+					}
+					EditorNode::get_singleton()->show_warning(TTR("The selected resource can't be used while opened in the script editor."));
+					return;
+				}
+			}
```

Comment:
What if it's already selected, will everything break? Picker doesn't use resources, it allows you to edit them. And it should always be able to display them, because they can be a pre-existing value in some object.

Reply:
And same question would apply to the scenes. What if the scene is already opened? What if we open a scene as a scene, then open it as text, and then close and restart the editor. What would happen when we try to restore opened scenes and files?

I'm not sure this approach works. It feels more like a strange band-aid.

Reply:
If the resource is already selected nothing seems to break. I'll use the word picked of used. So it says "The selected resource can't be picked while opened in the script editor" instead.

For the scene nothing also seems to break even when restarting the editor with the scene opened (first) and opened as text, but, for this commit it will trigger the error message I introduced even though the scene remains open as well as the text after pressing ok on the error.

Reply:
Sorry, my questions were more rhetorical. My overall point was that this warning will end up confusing users because depending on the order of steps it may work and it may not work. The fact that it breaks on project reload is a demonstration of such a confusing situation.

So I don't think this is a solution that we can go for.